crates/matrix-sdk/src/notification_settings/rules.rs

```diff
@@ -826,15 +862,11 @@ pub(crate) mod tests {
         let room_id = get_test_room_id();
         let mut rules = Rules::new(get_server_default_ruleset());
 
-        rules.delete_rules(&[(RuleKind::Room, room_id.to_string())], &[]).expect_err(
-            "A RemovePushRuleError is expected while trying to delete an unknown rule.",
-        );
```

Comment:
Now that I see this test has been removed, I think the `delete_rules` command construction should take into account the existing rules, to make sure we don't immediately try to remove a rule that wasn't there in the first place.